cli/root.go

```diff
@@ -698,14 +699,44 @@ func (r *RootCmd) createAgentClient() (*agentsdk.Client, error) {
 }
 
 // CurrentOrganization returns the currently active organization for the authenticated user.
-func CurrentOrganization(inv *clibase.Invocation, client *codersdk.Client) (codersdk.Organization, error) {
+func CurrentOrganization(r *RootCmd, inv *clibase.Invocation, client *codersdk.Client) (codersdk.Organization, error) {
+	conf := r.createConfig()
+	selected := ""
+	if conf.Organization().Exists() {
+		org, err := conf.Organization().Read()
+		if err != nil {
+			return codersdk.Organization{}, fmt.Errorf("read selected organization from config file %q: %w", conf.Organization(), err)
+		}
+		selected = org
+	}
+
+	// Verify the org exists and the user is a member
 	orgs, err := client.OrganizationsByUser(inv.Context(), codersdk.Me)
 	if err != nil {
-		return codersdk.Organization{}, nil
+		return codersdk.Organization{}, err
```

Comment:
@mafredri The unit test expecting `context cancelled` had to be changed because of this change. (https://github.com/coder/coder/pull/12259#discussion_r1500825567)

Before, if you failed to fetch the orgs, we'd return a nil uuid. I made this api failure fail the command now. So the unit tests fails differently now.

Reply:
I wouldn't expect `TestTemplateCreate/WithVariablesFileWithoutRequiredValue` to be testing for org failure though 🤔

Reply:
I didn't write the test, but there is a comment indicating this cli error is expected. We cancel the context before running the command.

https://github.com/coder/coder/blob/98666fd7d7d3e3449b4427d5f5ab6b8e8958765a/cli/templatecreate_test.go#L238-L244

Reply:
If we're testing against a cancelled context, that test is useless. I think it should either be removed or fixed. 😅 It's definitely not testing what the test name is indicating.